``7-base_geometry`` function
========================

Using ``7-base_geometry``
---------------------

Test Task 0

     >>> BaseGeometry = __import__('7-base_geometry').BaseGeometry


Section //
=========================

     >>> bg = BaseGeometry()
     >>> type(bg)
     <class '7-base_geometry.BaseGeometry'>

     >>> bg = BaseGeometry(None)
     Traceback (most recent call last):
     TypeError: object() takes no parameters

     >>> bg.area()
     Traceback (most recent call last):
     Exception: area() is not implemented

     >>> bg.integer_validator()
     Traceback (most recent call last):
     TypeError: integer_validator() missing 2 required positional arguments: 'name' and 'value'

     >>> bg.integer_validator(None)
     Traceback (most recent call last):
     TypeError: integer_validator() missing 1 required positional argument: 'value'

     >>> bg.integer_validator("testing", 0)
     Traceback (most recent call last):
     ValueError: testing must be greater than 0

     >>> bg.integer_validator("testing")
     Traceback (most recent call last):
     TypeError: integer_validator() missing 1 required positional argument: 'value'

     >>> bg.integer_validator("testing", "test")
     Traceback (most recent call last):
     TypeError: testing must be an integer

     >>> bg.integer_validator("testing", True)
     Traceback (most recent call last):
     TypeError: testing must be an integer

     >>> bg.integer_validator("testing", None)
     Traceback (most recent call last):
     TypeError: testing must be an integer

     >>> bg.integer_validator("testing", -7)
     Traceback (most recent call last):
     ValueError: testing must be greater than 0

     >>> bg.integer_validator("testing", (7, 9))
     Traceback (most recent call last):
     TypeError: testing must be an integer

     >>> bg.integer_validator("testing", [7])
     Traceback (most recent call last):
     TypeError: testing must be an integer

     >>> bg.integer_validator("testing", {7, 8})
     Traceback (most recent call last):
     TypeError: testing must be an integer

     >>> bg.integer_validator("testing", 7.7)
     Traceback (most recent call last):
     TypeError: testing must be an integer

     >>> bg.integer_validator("testing", 7)
